src/rust/engine/src/nodes.rs

```diff
@@ -1049,6 +1049,11 @@ impl Node for NodeKey {
   type Error = Failure;
 
   fn run(self, context: Context) -> NodeFuture<NodeResult> {
+    let start_timestamp = std::time::SystemTime::now()
+        .duration_since(std::time::SystemTime::UNIX_EPOCH)
+        .unwrap()
+        .as_secs() as f64;
```

Comment:
Since rust 1.33 (which we started using after you started putting this PR together), there are `Duration::as{millis,micros,nanos}`, so let's use one of those so we get sub-second granularity :)